# HG changeset patch
# User Arlo Breault <arlolra@gmail.com>
# Date 1414169680 25200
#      Fri Oct 24 09:54:40 2014 -0700
# Node ID 358de61ab92e106fe0ffd877aa71371ceae0b87f
# Parent  7e72c5322e5c082b2472e0b07a06627e222f3864
Prevent empty messages from falling back to the original

 * Irc will split up messages and empty lines fall back to the encrypted
   message in OTR.

diff --git a/chat/components/src/imConversations.js b/chat/components/src/imConversations.js
--- a/chat/components/src/imConversations.js
+++ b/chat/components/src/imConversations.js
@@ -29,17 +29,20 @@ function imMessage(aPrplMessage) {
 }
 imMessage.prototype = {
   __proto__: ClassInfo(["imIMessage", "prplIMessage"], "IM Message"),
   cancelled: false,
   color: "",
   _displayMessage: null,
 
   get displayMessage() {
-    return this._displayMessage || this.prplMessage.originalMessage;
+    // Explicitly test for null so that blank messages don't fall back to
+    // the original. Especially problematic in encryption extensions like OTR.
+    return this._displayMessage !== null ?
+      this._displayMessage : this.prplMessage.originalMessage;
   },
   set displayMessage(aMsg) { this._displayMessage = aMsg; },
 
   get message() this.prplMessage.message,
   set message(aMsg) { this.prplMessage.message = aMsg; },
 
   // from prplIMessage
   get who() this.prplMessage.who,
